qcschema/dev/properties/cc_properties.py

```diff
@@ -0,0 +1,45 @@
+"""
+The complete list of Coupled-Cluster properties.
+"""
+
+cc_properties = {}
+
+cc_properties["ccsd_same_spin_correlation_energy"] = {
+    "type":
+    "number",
+    "description":
+    "The unscaled portion of the CCSD correlation energy from same-spin or triplet doubles correlations."
+}
+
+cc_properties["ccsd_opposite_spin_correlation_energy"] = {
+    "type":
+    "number",
+    "description":
+    "The unscaled portion of the CCSD correlation energy from opposite-spin or singlet doubles correlations."
+}
+
+cc_properties["ccsd_singles_energy"] = {
+    "type": "number",
+    "description": "The singles portion of the CCSD correlation energy. Zero except in ROHF."
+}
+
+cc_properties["ccsd_doubles_energy"] = {
+    "type":
+    "number",
+    "description":
+    "The doubles portion of the CCSD correlation energy including same-spin and opposite-spin correlations."
+}
+
+cc_properties['ccsd_total_correlation_energy'] = {
```

Comment:
That seems reasonable to me. Should we also change `mp_properties.py` such that `mp2_total_correlation_energy` is renamed `mp2__correlation_energy`?

Reply:
Better get others' up/down votes on this one. I'd like to, but that's changing the schema, possibly in the database.

Reply:
It would cause the database to change, not particularly a big deal in the early days. I am partial to shortening this as well.

Reply:
Since the database owner isn't crying foul, I think go ahead and change it for both ccsd and mp2. For the moment, MolSSI has the only valuable instance with mp2 properties.

Reply:
Okay I have changed it and updated the test files that used those keywords. Presumably this will have cascading effects in QCEngine and elsewhere (?). How should we handle that?

Reply:
qcschema right now is a guidance document, so the cascade hasn't started :-)

Changing https://github.com/MolSSI/QCElemental/pull/54/files#diff-f178293574dc070baaac6713e52806beR35 is where the cascade starts. fortunately, few post-SCF calcs that are "keepers" yet run.

@dgasmith, (a) go ahead with the current PR, (b) wait a couple days for more to chime in, (c) expand to CCSD(T), CCSDT, CCSDTQ total/correlation energy, ccsd_iterations, and cc dipole?

Reply:
I could vote C if @sjrl is up for it. Otherwise we can do A.

Reply:
Yup I can do option C

Reply:
Okay I've added what was specified in option C). However, I can't get CCSDT and CCSDTQ information currently (Molpro doesn't have those), so I haven't made simple test jobs for them. Finally, should I add cc_dipole information to the `*_output.json` test jobs? I only ask because CC dipoles are usually only available if you are calculating a gradient or some other response property, it's not usually calculated when just calling energy (which is what is specified in the `*_input.json`).

Reply:
I wouldn't worry too much about examples for exotic properties at the moment. One concern on the `cc_*` properties is that it is possible to have the dipole moment at each level. Seems rare, but worth pondering.